fabric-climbable-v1/src/main/java/net/fabricmc/fabric/api/event/block/FallDeathSuffixCallback.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package net.fabricmc.fabric.api.event.block;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.util.Pair;
+
+public interface FallDeathSuffixCallback {
+
+	Event<FallDeathSuffixCallback> event = EventFactory.createArrayBacked(FallDeathSuffixCallback.class,
+		(fallDeathCallbacks -> (entity, state) -> {
+			String result = "generic";
+			int currentPriority = 0;
+
+			for (FallDeathSuffixCallback fallDeathCallback : fallDeathCallbacks) {
+				Pair<String, Integer> pair = fallDeathCallback.getFallDeathSuffix(entity, state);
```

Comment:
I would allow returning `null` if the listener is not interested in handling this situation. Avoids creating unnecessary objects. Also consider creating your own data class (eg. `FallDeathSuffixCallback.Result`) that takes a String and an int, to avoid boxing.